projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts

```diff
@@ -30,8 +30,16 @@ export class ProductImagesComponent {
     map(([, container]) => container)
   );
 
+  numberOfCarouselItems: number;
+
   constructor(private currentProductService: CurrentProductService) {}
 
+  ngOnInit() {
```

Comment:
instead of get the thumbs length in `ngOnInit`, you can add `tap` in:
```
thumbs$: Observable<any[]> = this.product$.pipe(
    map(product => this.createThumbs(product)),
    tap(thumbs => this.numberOfCarouselItems = thumbs.length)
  );
```